.github/workflows/common-api-ci.yml

```diff
@@ -0,0 +1,201 @@
+name: Shared Continuous Integration workflow for APIs.
+
+on:
+  workflow_call:
+    inputs:
+      # Path to the main API project.
+      api-project:
+        required: true
+        type: string
+      # Root name for the build artifact.
+      artifact-name:
+        required: true
+        type: string
+
+# Only run one instance of this workflow at a time per branch.
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  DOTNET_NOLOGO: true
+  NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+jobs:
+
+  test_build_release:
+    name: Test, Build, Publish on ${{ matrix.operating-system }}
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+        ## using latest LTS release - also it MUST be the SDK version.
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          global-json-file: global.json
+          source-url: "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json"
+
+      - name: Build with dotnet
+        run: dotnet build --configuration Release /WarnAsError
+
+      - name: Unit Tests (with Coverage)
+        run: |
+              # Skip code coverage test on Windows because it errors out after getting "confused" about the number of test projects.
+              if [ "$RUNNER_OS" == "Windows" ]; then
+                dotnet test
+              else
+                dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=lcov /p:CoverletOutput=../../lcov test/**
+              fi
+        shell: bash
+
+      - name: Save Code Coverage Output
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-${{ matrix.operating-system }}
+          path: lcov.info
+        if: matrix.operating-system != 'windows-latest'
+
+      - name: Download install tools
+        uses: actions/checkout@v4
+        with:
+          repository: nciocpl/nci.ocpl.api.shared
+          # Ref will need to be updated when version changes.
+          #ref: v2.x
+          # For development purposes.
+          ref: ticket/52-shared-wf
+          path: shared
+
+      - name: Publish
+        run: |
+            dotnet publish -c Release -o $GITHUB_WORKSPACE/out/${{ inputs.artifact-name }}-api ${{ inputs.api-project }}
+
+            ## directory for build-info.json
+            mkdir -p $GITHUB_WORKSPACE/out/${{ inputs.artifact-name }}-api/wwwroot/
+
+            # Copy deploy toool.
+            cp $GITHUB_WORKSPACE/shared/tools/deploy-api.ps1 $GITHUB_WORKSPACE/out/
+
+        shell: bash
+
+      - name: Record metadata
+        uses: nciocpl/build-info-action@v1.0.0
```

Comment:
Should this be v1.0.0? (really, we should probably do @v1, but that requires improving the tooling on that action to handle semver tags)

Reply:
Whoops!  No, that should indeed be v1.0.1.

(And there's a lot that could be done to improve that action and/or make it more useful.)